Automattic/wp-calypso
# Scheduled updates: Integrate fetching plugins per site ids and bind to plugins component

Related to https://github.com/Automattic/wp-calypso/issues/89473

## Proposed Changes

* Creates `useCoreSitesPluginsQuery` hook for fetching plugins by site IDs with caching logic since we need to ping the server for each site id
* Creates helper mapper methods
* Integrates`useCoreSitesPluginsQuery` query hook and binds data to the plugins selector component
* Adds validation message
* Adds logic for removing currently selected plugins if plugin disappears from the plugins list because of site selection update

## Testing Instructions

* Go to `/plugins/scheduled-updates/create`
* Check if the plugins selector component shows the unique union of plugins for the presented atomic sites

<img width="1341" alt="Screenshot 2024-04-16 at 22 54 14" src="https://github.com/Automattic/wp-calypso/assets/1241413/94392c8d-4815-4e24-a04b-4e2a64b24c6f">

<img width="1245" alt="Screenshot 2024-04-18 at 13 50 08" src="https://github.com/Automattic/wp-calypso/assets/1241413/a1abee0d-503b-4c6f-bb20-0013cf496288">

![Screen Capture on 2024-04-18 at 17-38-24](https://github.com/Automattic/wp-calypso/assets/1241413/c79a0d9e-08a3-46c7-94c6-9690b20f604b)


## Pre-merge Checklist
